api/notd/token_ownership_processor.py

```diff
@@ -19,82 +20,86 @@ def __init__(self) -> None:
 
 class TokenOwnershipProcessor:
 
-    def __init__(self,retriever: Retriever):
+    def __init__(self,retriever: Retriever, lockManager: LockManager):
         self.retriever = retriever
+        self.lockManager = lockManager
 
     async def calculate_token_single_ownership(self, registryAddress: str, tokenId: str)-> RetrievedTokenOwnership:
-        tokenTransfers = await self.retriever.list_token_transfers(
-            fieldFilters=[
-                StringFieldFilter(fieldName=TokenTransfersTable.c.registryAddress.key, eq=registryAddress),
-                StringFieldFilter(fieldName=TokenTransfersTable.c.tokenId.key, eq=tokenId),
-            ],
-            orders=[Order(fieldName=TokenTransfersTable.c.blockNumber.key, direction=Direction.DESCENDING)],
-            limit=1,
-        )
-        if len(tokenTransfers) == 0:
-            raise NoOwnershipException()
-        latestTokenTransfer = tokenTransfers[0]
-        return RetrievedTokenOwnership(
-            registryAddress=registryAddress,
-            tokenId=tokenId,
-            ownerAddress=latestTokenTransfer.toAddress,
-            transferDate=latestTokenTransfer.blockDate,
-            transferValue=latestTokenTransfer.value,
-            transferTransactionHash=latestTokenTransfer.transactionHash,
-        )
-
-    async def calculate_token_multi_ownership(self, registryAddress: str, tokenId: str) -> List[RetrievedTokenMultiOwnership]:
-        ownerships: Dict[str, RetrievedTokenMultiOwnership] = {}
-        offset = 0
-        limit = 500
-        while True:
+        async with self.lockManager.with_lock(name='ownership-processor', timeoutSeconds=5, expirySeconds=60):
```

Comment:
you should use the registryAddress and tokenId in the lock (in the key i guess) because we should allow calcaulting for different tokens at the same time

Reply:
also, this should go in the manager, not the processor, because its not saving here